tooling/src/hypothesistooling/__main__.py

```diff
@@ -404,28 +404,88 @@ def update_django_versions():
 
 def update_pyodide_versions():
     vers_re = r"(\d+\.\d+\.\d+)"
-    all_versions = re.findall(
+    all_pyodide_build_versions = re.findall(
         f"pyodide_build-{vers_re}-py3-none-any.whl",  # excludes pre-releases
         requests.get("https://pypi.org/simple/pyodide-build/").text,
     )
-    for pyodide_version in sorted(
+    pyodide_build_version = sorted(
         # Don't just pick the most recent version; find the highest stable version.
-        set(all_versions),
+        set(all_pyodide_build_versions),
         key=lambda version: tuple(int(x) for x in version.split(".")),
         reverse=True,
-    ):
-        makefile_url = f"https://raw.githubusercontent.com/pyodide/pyodide/{pyodide_version}/Makefile.envs"
-        match = re.search(
-            rf"export PYVERSION \?= {vers_re}\nexport PYODIDE_EMSCRIPTEN_VERSION \?= {vers_re}\n",
-            requests.get(makefile_url).text,
+    )[0]
+
+    cross_build_environments_url = "https://raw.githubusercontent.com/pyodide/pyodide/refs/heads/main/pyodide-cross-build-environments.json"
+    cross_build_environments_data = requests.get(cross_build_environments_url).json()
+
+    # Find the latest stable release for the Pyodide runtime/xbuildenv that is compatible
+    # with the pyodide-build version we found
+    stable_releases = [
+        release
+        for release in cross_build_environments_data["releases"].values()
+        if "a" not in release["version"]
+        and "b" not in release["version"]
+        and "rc" not in release["version"]
+    ]
+
+    compatible_releases = []
+    build_parts = [int(x) for x in pyodide_build_version.split(".")]
+
+    for release in stable_releases:  # sufficiently large values
+        min_build_version = release.get("min_pyodide_build_version", "0.0.0")
+        max_build_version = release.get("max_pyodide_build_version", "999.999.999")
+
+        min_parts = [int(x) for x in min_build_version.split(".")]
+        max_parts = [int(x) for x in max_build_version.split(".")]
+
```

Comment:
let's pull out a helper function:

```python
def version_tuple(v: str) -> tuple[int, int, int]:
    return tuple(int(x) for x in v.split("."))  # type: ignore
```

and then all of this logic can lean on tuple-comparison semantics, and thus:

```python
if (
    version_tuple(min_build_version)
    <= version_tuple(pyodide_build_version)
    <= version_tuple(max_build_version
):
    compatible_releases.append(release)
```